pkg/connector/sso_group.go

```diff
@@ -155,6 +169,81 @@ func ssoGroupBuilder(region string, ssoClient *awsSsoAdmin.Client, identityStore
 	}
 }
 
+func (g *ssoGroupResourceType) Grant(ctx context.Context, principal *v2.Resource, entitlement *v2.Entitlement) (annotations.Annotations, error) {
+	if principal.Id.ResourceType != resourceTypeSSOUser.Id {
+		return nil, errors.New("baton-aws: only sso users can be added to a sso group")
+	}
+
+	entitlementAnnotations := annotations.Annotations(entitlement.Resource.Annotations)
+	groupTrait := &v2.GroupTrait{}
+	ok, err := entitlementAnnotations.Pick(groupTrait)
+	if err != nil {
+		return nil, fmt.Errorf("baton-aws: error unmarshalling entitlement group trait: %w", err)
+	}
+	if !ok {
+		return nil, errors.New("baton-aws: entitlement missing group trait")
+	}
+	groupID, ok := resourceSdk.GetProfileStringValue(groupTrait.Profile, "aws_group_id")
```

Comment:
Is this group ID different from the return value of `ssoGroupIdFromARN`? It seems like that with the ARN of the group(and the user below) we could either make an API call to fetch teh group here and grab the ID, instead of relying on populated profile data.